linux_docker_resources/Dockerfile

```diff
@@ -38,6 +38,9 @@ RUN apt-get update && apt-get install -y build-essential ccache cmake pkg-config
 # Install build and test dependencies of ROS 2 packages.
 RUN apt-get update && apt-get install -y clang-format-3.8 cppcheck git pydocstyle pyflakes python3-coverage python3-flake8 python3-mock python3-nose python3-pep8 uncrustify
 
+# Install build dependencies for turtlebot demo.
+RUN if test -n ${INSTALL_TURTLEBOT2_DEMO_DEPS}; apt-get update && apt-get install -y libeigen3-dev libtinyxml-dev
```

Comment:
I think you will also want to move this to after [this line](https://github.com/ros2/ci/commit/a5ff6c4e14dbe70d829676f5aeb533a15ff3bab2#diff-a7a7037d3453364950fd80f5f91ed7eeR87) that prints a different string each day, so that (1) the docker images only diverge once necessary and (2) the docker image gets invalidated daily. this ensures that "fresh" dependencies get installed.

Otherwise, if the _text_ in the dockerfile of what dependencies to install does not change, it will not get re-built even if there are newer versions of dependencies that ought to be installed. For system dependencies it might not matter much but we will probably end up installing ROS packages, which get released regularly. (@dirk-thomas please chime in if this doesn't match what you explained to me the other day!)